app/src/main/java/com/minimalisticapps/priceconverter/presentation/ui/widget/TextInputBtc.kt

```diff
@@ -1,103 +1,36 @@
 package com.minimalisticapps.priceconverter.presentation.ui.widget
 
 import android.annotation.SuppressLint
-import android.util.Log
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.text.KeyboardOptions
 import androidx.compose.material.TextField
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.focus.onFocusChanged
-import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.focus.onFocusEvent
 import androidx.compose.ui.text.input.KeyboardType
 import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
-import com.minimalisticapps.priceconverter.common.utils.formatBtc
-import com.minimalisticapps.priceconverter.common.utils.parseBigDecimalFromString
 import com.minimalisticapps.priceconverter.presentation.home.viewmodels.HomeViewModel
 
 @SuppressLint("UnrememberedMutableState")
 @Composable
 fun TextInputBtc(
     homeViewModel: HomeViewModel = hiltViewModel(),
 ) {
-    val count = remember {
-        mutableStateOf(0)
-    }
-
     TextField(
         modifier =
         Modifier
             .fillMaxWidth()
             .padding(start = 15.dp, end = 15.dp)
-            .onFocusChanged {
+            .onFocusSelectAll(homeViewModel.textFiledValueBtc)
+            .onFocusEvent {
                 if (it.isFocused) {
-                    homeViewModel.setTextFieldValueBtc(
-                        homeViewModel.textFieldValueBtc.value.copy(
-                            selection = TextRange(
-                                0,
-                                homeViewModel.textFieldValueBtc.value.text.length
-                            )
-                        )
-                    )
-                    count.value = 1
-                } else {
-                    count.value = 0
+                    homeViewModel.selectedCoin.value = "BTC"
                 }
             },
-        value = homeViewModel.textFieldValueBtc.value,
-        onValueChange = { textFieldValue ->
```

Comment:
removed spaggeti code, now the formatting logic for commas is in separate method that is tested